app/src/main/kotlin/com/wire/android/ui/sharing/ImportMediaScreen.kt

```diff
@@ -0,0 +1,207 @@
+import com.wire.android.util.extension.getActivity
+import kotlinx.collections.immutable.persistentMapOf
+import kotlinx.coroutines.flow.SharedFlow
+
+@Composable
+fun ImportMediaScreen(importMediaViewModel: ImportMediaViewModel = hiltViewModel()) {
+    val context = LocalContext.current
+    val searchBarState = rememberSearchbarState()
+    val snackBarHostState: SnackbarHostState = remember { SnackbarHostState() }
+    LaunchedEffect(importMediaViewModel.importMediaState) {
+        if (importMediaViewModel.importMediaState.importedAssets.isEmpty()) {
+            context.getActivity()?.let { importMediaViewModel.handleReceivedDataFromSharingIntent(it) }
+        }
+    }
+
+    ImportMediaContent(searchBarState, snackBarHostState, importMediaViewModel)
+}
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalPagerApi::class, ExperimentalFoundationApi::class)
+@Composable
+fun ImportMediaContent(searchBarState: SearchBarState, snackbarHostState: SnackbarHostState, importMediaViewModel: ImportMediaViewModel) {
+    val actionButtonTitle = stringResource(R.string.import_media_send_button_title)
+    Scaffold(
+        topBar = {
+            WireCenterAlignedTopAppBar(
+                elevation = 0.dp,
+                onNavigationPressed = importMediaViewModel::navigateBack,
+                title = stringResource(id = R.string.import_media_content_title),
+                actions = {
+                    UserProfileAvatar(
+                        avatarData = UserAvatarData(importMediaViewModel.importMediaState.avatarAsset),
+                        clickable = remember { Clickable(enabled = false) { } }
+                    )
+                }
+            )
+        },
+        snackbarHost = {
+            SwipeDismissSnackbarHost(
+                hostState = snackbarHostState,
+                modifier = Modifier.fillMaxWidth()
+            )
+        },
+        modifier = Modifier.background(colorsScheme().background),
+        content = { internalPadding ->
+            val importedItemsList: List<ImportedMediaAsset> = importMediaViewModel.importMediaState.importedAssets
+            val itemsToImport = importedItemsList.size
+            val pagerState = rememberPagerState()
+            val isMultipleImport = itemsToImport > 1
+
+            Column(
+                modifier = Modifier
+                    .padding(internalPadding)
+                    .fillMaxSize()
+            ) {
+                val horizontalPadding = dimensions().spacing8x
+                val screenWidth = LocalConfiguration.current.screenWidthDp.dp
+                val itemWidth = if (isMultipleImport) dimensions().importedMediaAssetSize else screenWidth - (horizontalPadding * 2)
+                val contentPadding = PaddingValues(start = horizontalPadding, end = (screenWidth - itemWidth + horizontalPadding))
+                val lazyListState = rememberLazyListState()
+                if (importMediaViewModel.importMediaState.isImporting) {
```

Comment:
to not repeat your self each time with importMediaViewModel.importMediaState you can enlocse content of this COmposagble within with(importMediaViewmodel.importMediaState) {

}